streambed/src/lib.rs

```diff
@@ -131,52 +131,56 @@ pub async fn get_secret_value(
 }
 
 /// Given a secret store, a path to a secret, and a byte buffer to be decrypted,
-/// decrypt it in place. Returns a JSON decoded structure if decryption
+/// decrypt it in place. Returns a decoded structure if decryption
 /// was successful.
 /// The secret is expected to reside in a data field named "value" and
 /// is encoded as a hex string of 32 characters (16 bytes)
 /// The buffer is expected to contain both the salt and the bytes to be decrypted.
-pub async fn decrypt_buf<'a, T>(
+pub async fn decrypt_buf<'a, T, D>(
     ss: &impl secret_store::SecretStore,
     secret_path: &str,
     buf: &'a mut [u8],
```

Comment:
I know you've already merged this but do `buf` and the slice arg in `D` need to be `mut` here? `serde_json::from_slice` doesn't seem to need the argument to be `mut`.

Reply:
I'm actually re-working this PR so it is never too late! Some deserialisers do so in place, hence the mut.

Reply:
Ok fair enough!

Reply:
Actually, the deserialize closure doesn't need to mut - but the buf passed in still does. I'll update the signature to reflect this and push another commit. Thanks.